src/common/models/transaction-details.model.ts

```diff
@@ -15,28 +16,29 @@ export class TransactionDetailsModel {
   readonly status: string;
 
   @ApiProperty({ description: 'Transaction fees', type: FeesModel })
+  @Type(() => FeesModel)
   readonly fees: FeesModel;
 
   @ApiProperty({
     type: 'boolean',
     example: true,
-    description:
-      'Not all transaction can be subsidized. Can be used to check if the transaction can be subsidized',
+    description: 'Indicates if the transaction can be subsidized',
   })
   readonly supportsSubsidy: boolean;
 
   @ApiProperty({
     description: 'Paying account details',
     type: PayingAccountModel,
   })
+  @Type(() => PayingAccountModel)
   readonly payingAccount: PayingAccountModel;
 
   constructor({ status, fees, supportsSubsidy, payingAccount }: TransactionDetailsModel) {
     Object.assign(this, {
       status,
       supportsSubsidy,
-      payingAccount: new PayingAccountModel(payingAccount),
-      fees: new FeesModel(fees),
+      payingAccount,
+      fees,
```

Comment:
Since all of the properties are directly used, you can skip the destructing part here
```
constructor(model: TransactionDetailsModel) {
  Object.assign(this, model);
 }
 ```